workbench/src/preload/api.js

```diff
@@ -0,0 +1,66 @@
+import crypto from 'crypto';
+
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { ipcRenderer } from 'electron';
+
+import { ipcMainChannels } from '../main/ipcMainChannels';
+import { getLogger } from '../main/logger';
+
+const logger = getLogger();
+
+// Most IPC initiates in renderer and main does the listening,
+// but these channels are exceptions: renderer listens for them
+const ipcRendererChannels = [
+  /invest-logging-*/,
+  /invest-stdout-*/,
+  /invest-exit-*/,
+  /download-status/,
+];
+
+export default {
+  // Port where the flask app is running
+  PORT: process.env.PORT,
+  // Workbench logfile location, so Report window can open to it
+  LOGFILE_PATH: logger.transports.file.getFile().path,
+  // The gettext callable; a partially applied function
+  _: ipcRenderer.sendSync.bind(null, ipcMainChannels.GETTEXT),
+  getLogger: getLogger,
+  crypto: {
+    createHash: (algorithm) => crypto.createHash(algorithm)
+  },
+  electron: {
+    ipcRenderer: {
```

Comment:
Do we have `electron` here as a nested level to help with maintaining understanding in the renderer code?

Reply:
I figured this way it matches the actual `electron` API, where these methods are available at the `electron` namespace. Also it helps keep things organized right here since there's other non-electron methods on `Workbench`.